subiquity/common/types.py

```diff
@@ -394,6 +394,20 @@ class UbuntuAdvantageInfo:
     token: str
 
 
+class UbuntuAdvantageCheckTokenStatus(enum.Enum):
+    VALID_TOKEN = enum.auto()
+    INVALID_TOKEN = enum.auto()
+    EXPIRED_TOKEN = enum.auto()
+    UNKNOWN_ERROR = enum.auto()
+
+
+@attr.s(auto_attribs=True)
+class UbuntuAdvantageCheckTokenAnswer:
+    status: UbuntuAdvantageCheckTokenStatus
+
+    services: Optional[List[dict]]
```

Comment:
Is 'dict' all we want to say about the contents here? I guess it's just want comes out of the ua binary so perhaps it's not worth saying overmuch. But afaict the client only looks for the description, so maaybe this could just be a list of descriptions?

Not going to insist on any change here.

Reply:
I have been originally hesitant about this one ; namely because I didn't know what information would be valuable for the user. The description makes sense because we display it but I wouldn't be surprised if the name would also be useful as an identifier e.g. maybe at some point we want to do:

```python
is_livepatch_activable = any([service["name"] == "livepatch" for service in services])
```

I don't know if it makes sense to keep the other keys around:
* entitled -> at this point we know it's true
* auto_enabled -> TBH, I don't know what this is ; I should probably check
* available -> at this point we know it's true

I'd be happy to change but I'm not sure what direction to take yet.

Reply:
Using a `typing.TypedDict` would probably make sense.
```python
from typing import TypedDict
class ServiceDict(TypedDict):
    name: str
    description: str
```
And then we'd need to assert the presence of these two keys (if they're the ones we want to include) after calling `u-a-c`

Reply:
Is there a difference between TypedDict and an attr.s class for this sort of thing really? But that doesn't support extra keys either. One approach would be to do some filtering in check_token_GET to only the keys that the api defines I guess. Or just leave this as is for now :)

Speaking generally I don't want us to wast time thinking too hard about "what might be useful in the future" for api stuff. We have very tight bindings between the clients and servers and that's fine -- if needs change, the API changes :)

Reply:
I'll go with the `attr.s` + filtering. I think it makes the most sense. Thanks

Reply:
I decided to filter the keys of the services soon after calling `u-a-c` (i.e. just after filtering out the services that are not activable). Because some unit tests were expecting the "entitled" & "available" keys to still be around, I had to rework the unit tests a little.